src/application/services/tree-calculation-handler.service.ts

```diff
@@ -0,0 +1,115 @@
+import { Result } from 'rich-domain';
+import { differenceBy, isEmpty, isEqual } from 'lodash';
+import { TreeCalculationService } from 'src/domain/services/tree-calculation/tree-calculation.service';
+import { BasicFolderChange } from '../interfaces/basic-folder-changes.interface';
+import { RomachRepositoryInterface } from '../interfaces/romach-repository.interface';
+import { RomachEntitiesApiInterface } from '../interfaces/romach-entities-api.interface';
+import { RetryUtils } from 'src/utils/RetryUtils/RetryUtils';
+import { BasicFolder } from 'src/domain/entities/BasicFolder';
+import { Hierarchy } from 'src/domain/entities/Hierarchy';
+import { AppLoggerService } from 'src/infra/logging/app-logger.service';
+
+export interface TreeCalculationHandlerServiceOptions {
+    logger: AppLoggerService,
+    romachRepository: RomachRepositoryInterface,
+    treeCalculationService: TreeCalculationService,
+    romachEntitiesApiInterface: RomachEntitiesApiInterface,
+    maxRetry: number
+}
+
+export class TreeCalculationHandlerService {
+    constructor(private options: TreeCalculationHandlerServiceOptions) { }
+
+    async execute(changes: BasicFolderChange): Promise<Result<void>> {
+        const currentFoldersFromRepositoryResult = await this.getCurrentFoldersFromRepository();
+        const currentHierarchiesFromRepositoryResult = await this.getCurrentHierarchiesFromRepository();
```

Comment:
why to read the Hierarchies before you know that you need to recalculate the tree?